### PR TITLE
Remove deprecated 'rule' command

### PR DESCRIPTION
**Pull request by gpetiot**
Wait for 2.0 before merging this

### CODE REVIEW

**Comment by gpetiot** [comment]
What is the situation regarding the `#require` directive? It's weird we didn't have these regressions before rebasing this branch

**Comment by NathanReb** [comment]
> What is the situation regarding the #require directive?

What do you mean by that? If you're asking whether we still want to support them then the answer is definitely yes!

Not sure why those regression showed up but I'll look into it once we have compat with 4.13 merged!

**Comment by gpetiot** [comment]
On last test is KO on some versions (no idea why some are passing but that's not the purpose), `mdx-test-require-local-package`, @NathanReb do you remember the purpose of this test and if it's still makes sense after the rule command is removed?

File: test/bin/misc-test-cases/mdx-test-require-local-package/dune [hunk 1]
```diff
@@ -1,9 +1,9 @@
 (rule
- (targets dune.inc.gen)
+ (alias runtest)
  (deps
-  (package mdx))
```
Comment (NathanReb) [inline test/bin/misc-test-cases/mdx-test-require-local-package/dune:4]: Why is this removed? This is required because of the `ocaml-mdx` and `ocaml-mdx-test` binaries trick.

**Comment by NathanReb** [comment]
Have you tried without the `%{bin:...}` change. That's the only thing I can think of here.

It looks like it's looking for the other binary in the wrong place or that it's not built.

**Comment by NathanReb** [comment]
Ah wait, was that the only failing test? In that case that's perfectly normal and it's because it needs the `(package mdx)` dep, as mentioned in the comment above!

**Comment by gpetiot** [comment]
> Have you tried without the `%{bin:...}` change. That's the only thing I can think of here.
> 
> It looks like it's looking for the other binary in the wrong place or that it's not built.

I reverted this, it can be done in another PR anyway.

File: lib/block.ml [hunk 2]
```diff
@@ -215,32 +211,6 @@ let set_variables t = t.set_variables
 
 let unset_variables t = t.unset_variables
 
-let explicit_required_packages t = t.required_packages
-
-let require_re =
-  let open Re in
-  seq [ str "#require \""; group (rep1 any); str "\"" ]
-
-let require_from_line line =
```
Comment (Leonidas-from-XIV) [inline lib/block.ml:224]: Since I am a bit new to this project I'm a bit confused. I don't understand what removing the rule command has to do with removing support for `require`? Or am I misunderstanding this?

**Review by Leonidas-from-XIV — COMMENTED** [review]
The code looks fine but I am confused why this is removing so much code.

File: lib/block.ml (hunk 2 above)
Comment (gpetiot) [reply lib/block.ml:224]: This code was only used to set up the `deps` field of dune rules (as part of the rule command) and it doesn't impact the interpretation of `require` directive in the blocks

**Review by Leonidas-from-XIV — APPROVED** [review]
